docs/tutorials/install_and_admin/geonode_install/install_geonode_application.txt

```diff
@@ -1,120 +1,126 @@
-    $ sudo apt-get install build-essential openssh-server gettext nano vim unzip zip patch git-core postfix
-
-    $ sudo apt install -y openjdk-8-jre openjdk-8-jdk
-    $ sudo update-java-alternatives --set java-1.8.0-openjdk-amd64
-    $ export JAVA_HOME=$(readlink -f /usr/bin/java | sed "s:bin/java::")
-    $ export PATH=$JAVA_HOME'bin/java':$PATH
-
-    $ sudo apt-add-repository ppa:ubuntugis && sudo apt-get update && sudo apt-get upgrade
-    $ sudo apt-add-repository ppa:ubuntugis/ppa && sudo apt-get update && sudo apt-get upgrade
-    $ sudo apt-get install gcc apache2 libapache2-mod-wsgi libgeos-dev libjpeg-dev libpng-dev libpq-dev libproj-dev libxml2-dev libxslt-dev
-    $ sudo apt-add-repository ppa:ubuntugis/ubuntugis-testing && sudo apt-get update && sudo apt-get upgrade
-    $ sudo apt-get install gdal-bin libgdal20 libgdal-dev
-    $ sudo apt-get install python-gdal python-pycurl python-imaging python-pastescript python-psycopg2 python-urlgrabber
-    $ sudo apt-get install postgresql postgis postgresql-9.5-postgis-scripts postgresql-contrib
-    $ sudo apt-get install tomcat8
-
-    $ sudo apt-get update && sudo apt-get upgrade && sudo apt-get autoremove && sudo apt-get autoclean && sudo apt-get purge && sudo apt-get clean
-
-.. image:: img/install_packages.png
-   :width: 600px
-   :alt: Install Packages
-
-.. note:: If you will be prompted for `geo` user's password (`geo`) and for confirmation twice
-
-    .. image:: img/confirm_Install.png
-       :width: 600px
-       :alt: Confirm Installation
-
-.. warning:: The installation process is going to take several minutes and it will need to download packages from Internet.
-
-At this point we have all the packages we need on the system.
-
-GeoNode Setup
-=============
-First of all we need to prepare a new Python Virtual Environment:
+    sudo -i apt update
+    sudo apt install openjdk-8-jdk-headless default-jdk-headless -y
+    sudo update-java-alternatives --set /usr/lib/jvm/java-1.8.0-openjdk-amd64
+
+    sudo apt install apache2 \
+        build-essential \
+        default-jdk \
+        gcc \
+        gdal-bin \
+        gettext \
+        git \
+        git-core \
+        libapache2-mod-wsgi \
+        libgdal-dev \
+        libgdal20 \
+        libgeos-dev \
+        libjpeg-dev \
+        libpng-dev \
+        libpq-dev \
+        libproj-dev \
+        libxml2 \
+        libxml2-dev \
+        libxslt-dev \
+        libxslt1-dev \
+        nano \
+        openssh-server \
+        patch \
+        postfix \
+        postgis \
+        postgresql \
+        postgresql-10-postgis-2.4-scripts \
+        postgresql-contrib \
+        python-dev \
+        python-gdal \
+        python-pastescript \
+        python-pil \
+        python-pip \
+        python-psycopg2 \
+        python-pycurl \
+        python-urlgrabber \
+        python-virtualenv \
+        software-properties-common \
+        tomcat8 \
+        unzip \
+        vim \
+        zip \
+        zlib1g-dev
```

Comment:
Wouldn't be better to keep these dependencies in a different file instead using the documentation? I would be nice not to duplicate lines between setup procedures, CI and documentation.

Reply:
Yes! This helps to keep it manageable! Thanks.

Reply:
@frafra @afabiani I like the idea. What do you think about adding a file called apt-packages.txt to the geonode-core root directory. As Francesco says this can be reused for CI. As these docs target the use of geonode project we could explain the package install with something like:

```
# save the file path by use of $VIRTUAL_ENV where geonode core is installed to
apt_packages="$VIRTUAL_ENV/src/geonode/apt-packages.txt"

# Install packages line by line ignoring comments (untested)
apt-get install $(grep -vE "^\s*#" $aptpackages | tr "\n" " ")
```

Further, give some explanation what happens here.